### ISSUE TITLE
[Arm64] GcStress0x4: NullableTest45::BoxUnboxToQGen<MixedAllStruct> fails

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/coreclr#10454]
These tests fail in NullableTest45::BoxUnboxToQGen&lt;MixedAllStruct&gt; under GCStress0x4
```
JIT/Directed/nullabletypes/castclassvaluetype_r/castclassvaluetype_r.sh
JIT/Directed/nullabletypes/castclassvaluetype_d/castclassvaluetype_d.sh
JIT/Directed/nullabletypes/castclassvaluetype_do/castclassvaluetype_do.sh
```
The test had been fixed earlier by fixing #9655.  This is similar to the amd64 fix #2890 which fixed #2763..  

This is the beginning of the assembly...
```
; Assembly listing for method NullableTest45:BoxUnboxToNQGen(struct):bool
; Emitting BLENDED_CODE for 
; optimized code
; fp based frame
; fully interruptible
; Final local variable assignments
;
;  V00 arg0         [V00,T01] (  6,   8  )   byref  ->  x19        
;  V01 loc0         [V01,T08] (  2,   2  )  struct (112) [fp+0x1E0]   do-not-enreg[SB] must-init ld-addr-op
;  V02 loc1         [V02,T07] (  3,   2  )    bool  ->  [fp+0x1DC]   do-not-enreg[H]
;  V03 tmp0         [V03,T02] (  5,  10  )     ref  ->   x0        
;  V04 tmp1         [V04,T03] (  4,   6.5)     ref  ->   x1        
;  V05 tmp2         [V05,T05] (  5,   4.5)    long  ->   x9        
;  V06 tmp3         [V06,T00] (  9,  13.5)     ref  ->  x19        
;  V07 tmp4         [V07,T04] (  3,   6  )  struct (112) [fp+0x168]   do-not-enreg[SB] must-init
;  V08 tmp5         [V08    ] (  2,   4  )  struct (112) [fp+0xF8]   do-not-enreg[XSB] must-init addr-exposed
;* V09 tmp6         [V09    ] (  0,   0  )     ref  ->  zero-ref   
;  V10 tmp7         [V10    ] (  6,  12  )  struct (112) [fp+0x88]   do-not-enreg[XSB] must-init addr-exposed
;  V11 tmp8         [V11,T06] (  2,   4  )    long  ->   x8        
;  V12 tmp9         [V12    ] (  3,   6  )  struct (112) [fp+0x18]   do-not-enreg[XSB] must-init addr-exposed
;# V13 OutArgs      [V13    ] (  1,   1  )  lclBlk ( 0) [sp+0x00]  
;  V14 PSPSym       [V14    ] (  1,   1  )    long  ->  [fp+0x250]   do-not-enreg[X] addr-exposed
;
; Lcl frame size = 584

G_M25826_IG01:
        D10043FF          sub     sp, sp, #16
        F90007F3          str     x19, [sp,#8]
        D10943FF          sub     sp, sp, #592
        A9007BFD          stp     fp, lr, [sp]
        910003FD          mov     fp, sp
        910063A1          add     x1, fp, #24
        D2800462          mov     x2, #35
        A8817C3F          stp     xzr, xzr, [x1],#16
        F1000442          subs    x2, x2, #1
        54FFFFC8          bhi     pc-12 (-3 instructions)
        F900003F          str     xzr, [x1]
        910983E1          add     x1, sp, #608
        F9012BA1          str     x1, [fp,#592] // [V14 PSPSym]
        AA0003F3          mov     x19, x0

G_M25826_IG02:
        D29AD500          mov     x0, #0xd6a8
        F2A44D00          movk    x0, #0x2268 LSL #16
        F2C00FE0          movk    x0, #127 LSL #32
        97FFEF0D          bl      CORINFO_HELP_NEWSFAST
        9100200E          add     x14, x0, #8
        AA1303ED          mov     x13, x19
        97FFEF12          bl      CORINFO_HELP_ASSIGN_BYREF
        97FFEF11          bl      CORINFO_HELP_ASSIGN_BYREF
        97FFEF10          bl      CORINFO_HELP_ASSIGN_BYREF
        97FFEF0F          bl      CORINFO_HELP_ASSIGN_BYREF
        F84085AC          ldr     x12, [x13],#8
        F80085CC          str     x12, [x14],#8
```

Issue occurs if GCStressStart is set to trigger on or before the CORINFO_HELP_NEWSFAST call.

Related GCInfo begining follows.

```
; Total bytes of code 852, prolog size 56 for method NullableTest45:BoxUnboxToNQGen(struct):bool
; ============================================================
Set code length to 852.
Set ReturnKind to Scalar.
Set stack base register to fp.
Set PSPSym stack slot to -16.
Set WantsReportOnlyLeaf.
Set Outgoing stack arg area size to 0.
Stack slot id for offset 480 (0x1e0) (frame) (untracked) = 0.
Stack slot id for offset 488 (0x1e8) (frame) (untracked) = 1.
Stack slot id for offset 496 (0x1f0) (frame) (untracked) = 2.
Stack slot id for offset 504 (0x1f8) (frame) (untracked) = 3.
Stack slot id for offset 360 (0x168) (frame) (untracked) = 4.
Stack slot id for offset 368 (0x170) (frame) (untracked) = 5.
Stack slot id for offset 376 (0x178) (frame) (untracked) = 6.
Stack slot id for offset 384 (0x180) (frame) (untracked) = 7.
Stack slot id for offset 248 (0xf8) (frame) (untracked) = 8.
Stack slot id for offset 256 (0x100) (frame) (untracked) = 9.
Stack slot id for offset 264 (0x108) (frame) (untracked) = 10.
Stack slot id for offset 272 (0x110) (frame) (untracked) = 11.
Stack slot id for offset 136 (0x88) (frame) (untracked) = 12.
Stack slot id for offset 144 (0x90) (frame) (untracked) = 13.
Stack slot id for offset 152 (0x98) (frame) (untracked) = 14.
Stack slot id for offset 160 (0xa0) (frame) (untracked) = 15.
Stack slot id for offset 24 (0x18) (frame) (untracked) = 16.
Stack slot id for offset 32 (0x20) (frame) (untracked) = 17.
Stack slot id for offset 40 (0x28) (frame) (untracked) = 18.
Stack slot id for offset 48 (0x30) (frame) (untracked) = 19.
Register slot id for reg x19 (byref) = 20.
Register slot id for reg x0 = 21.
Register slot id for reg x14 (byref) = 22.
Register slot id for reg x13 (byref) = 23.
Register slot id for reg x1 = 24.
Register slot id for reg x19 = 25.
Register slot id for reg x12 = 26.
Register slot id for reg x0 (byref) = 27.
Register slot id for reg x8 (byref) = 28.
Register slot id for reg x1 (byref) = 29.
Set state of slot 20 at instr offset 0x38 to Live.
Set state of slot 21 at instr offset 0x48 to Live.
Set state of slot 22 at instr offset 0x4c to Live.
```
```
Defining interruptible range: [0x38, 0x31c).
Defining interruptible range: [0x340, 0x348).
```

The arg0/X0 is a BYREF which is copied to X19 at offset 0x34.  X0 should be dead at 0x38, then live at 0x48.  This looks consistnet with the GCInfo.

I have not found the root cause of the failure or identified X0 as the relevant GCRef.  It is just a working guess at this time.

I put this on hold pending other higher priority work.

**Comment by sdmaclea** [comment]
@jashook Please assign this to me

**Comment by sdmaclea** [comment]
Fixed by #10534
